Dockerfile

```diff
@@ -0,0 +1,155 @@
+ARG JUPYTER_MINIMAL_VERSION=lab-3.3.2@sha256:a4bf48221bfa864759e0f248affec3df1af0a68ee3e43dfc7435d84926ec92e8
+FROM jupyter/minimal-notebook:${JUPYTER_MINIMAL_VERSION} as base
+
+
+LABEL maintainer="iavarone"
+
+ENV JUPYTER_ENABLE_LAB="yes"
+# autentication is disabled for now
+ENV NOTEBOOK_TOKEN=""
+ENV NOTEBOOK_BASE_DIR="$HOME/work"
+
+USER root
+
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends \
+  ffmpeg \
+  make \
+  dvipng \
+  gosu \
+  octave \
+  gnuplot \
+  liboctave-dev \
+  bc \
+  ghostscript \
+  texlive-xetex \
+  texlive-fonts-recommended \
+  texlive-latex-recommended \
+  texlive-fonts-extra \
+  zip \
+  fonts-freefont-otf \
+  && \
+  apt-get clean && rm -rf /var/lib/apt/lists/*   
+
+RUN pip --no-cache --quiet install --upgrade \
+  pip \
+  setuptools \
+  wheel
+
+# Service (dakota) specific installation
+# --------------------------------------------------------------------
+
+WORKDIR /opt
+
+# Dependencies for compilation
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends \
+  gcc \
+  cmake \
+  libboost-all-dev \
+  libblas-dev \
+  liblapack-dev \
+  libopenmpi-dev \
+  openmpi-bin \
+  gsl-bin \
+  libgsl-dev \
+  perl \
+  libhdf5-dev \
+  gfortran
+
+# Download dakota tar
```

Comment:
why not using a multistage Dockerfile? You can have a stage to build dakota and then copy it in the final image. This way you avoid having all the build tools in the final image.

Reply:
I made an attempt to a multistage Dockerfile (I used as an example the [osparc-python-runner](https://github.com/ITISFoundation/osparc-python-runner/blob/master/docker/python/Dockerfile)).

I realized afterwards that some of those build tools are actually dependencies to run the library (so I moved them to the base stage). There's still margin for improvement here.